applications/GeoMechanicsApplication/custom_elements/line_2d_5.hpp

```diff
@@ -0,0 +1,1314 @@
+            const override
+        {
+            Matrix J(2, 1);
+            this->Jacobian(J, IntegrationPointIndex, ThisMethod);
+            return std::sqrt(std::pow(J(0, 0), 2) + std::pow(J(1, 0), 2));
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * @brief Determinant of jacobian in given point. This method calculate determinant of jacobian matrix in given point.
+         * @param rPoint point which determinant of jacobians has to be calculated in it.
+         * @return Determinamt of jacobian matrix \f$ |J| \f$ in given point.
+         * @see DeterminantOfJacobian
+         * @see InverseOfJacobian
+         */
+        double DeterminantOfJacobian(const CoordinatesArrayType& rPoint) const override
+        {
+            Matrix J(2, 1);
+            this->Jacobian(J, rPoint);
+            return std::sqrt(std::pow(J(0, 0), 2) + std::pow(J(1, 0), 2));
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /** EdgesNumber
+         * @return SizeType containes number of this geometry edges.
+         */
+        SizeType EdgesNumber() const override
+        {
+            return 4;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /** FacesNumber
+         * @return SizeType containes number of this geometry edges/faces.
+         */
+        SizeType FacesNumber() const override
+        {
+            return EdgesNumber();
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        ///@}
+        ///@name Shape Function
+        ///@{
+        /**
+         * @brief This method gives all non-zero shape functions values evaluated at the rCoordinates provided
+         * @note There is no control if the return vector is empty or not!
+         * @return Vector of values of shape functions \f$ F_{i} \f$ where i is the shape function index (for NURBS it is the inde of the local enumeration in the element).
+         * @see ShapeFunctionValue
+         * @see ShapeFunctionsLocalGradients
+         * @see ShapeFunctionLocalGradient
+         */
+        Vector& ShapeFunctionsValues(Vector& rResult, const CoordinatesArrayType& rCoordinates) const override
+        {
+            if (rResult.size() != 5) rResult.resize(5, false);
+            //
+            double xcor = rCoordinates[0];
```

Comment:
Please use "const" if the variable will not change later.

Reply:
Done